pkg/promqlgen/promqlgen.go

```diff
@@ -64,3 +61,53 @@ func LabelSelectorsToRelabelConfig(matches []string) (*monv1.RelabelConfig, erro
 		Regex:        regex,
 	}, nil
 }
+
+func GroupLabelSelectors(matches []string) (string, error) {
+	labelSets, err := parseMetricSelectorFromArray(matches)
+	if err != nil {
+		return "", nil
+	}
+	newLabelSet := map[string][]string{}
+	for _, ls := range labelSets {
+		for _, lm := range ls {
+			_, exists := newLabelSet[lm.Name]
+			if exists {
+				newLabelSet[lm.Name] = append(newLabelSet[lm.Name], lm.Value)
+			} else {
+				newLabelSet[lm.Name] = []string{lm.Value}
+			}
+		}
+
+	}
+
+	keys := make([]string, 0, len(newLabelSet))
+	for k := range newLabelSet {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	res := "{"
+	i := 0
+	for _, k := range keys {
+		res += k + `=~"`
+		res += strings.Join(newLabelSet[k], "|") + `"`
```

Comment:
We need to add a trailing `|` to the label selectors otherwise it will filter too many timeseries. Which means `{__name__=~"ALERTS|node_uname_info|csv_abnormal|",alertstate=~"firing|"}` instead of `{__name__=~"ALERTS|node_uname_info|csv_abnormal",alertstate=~"firing"}`.

Reply:
For some reason, probably something obvious:
`{__name__=~"ALERTS|node_uname_info|csv_abnormal|",alertstate=~"firing|"}` 
does not work, but promql says:
> Error executing query: invalid parameter ‘query’: 1:1498: parse error: vector selector must contain at least one non-empty matcher

`{__name__=~"ALERTS|node_uname_info|csv_abnormal",alertstate=~"firing|"}` instead did this and it seems to produce the correct number.

Reply:
Right, removing the extra `|` in the `__name__` selector as you suggest is the right thing to do.